File: src/main/java/com/pancisin/bookster/models/Locale.java
package com.pancisin.bookster.models;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonView;
import com.pancisin.bookster.models.views.Compact;
import com.pancisin.bookster.models.views.Summary;

@Entity
@Table(name = "locales")
public class Locale {

	@Id
	@JsonView(Compact.class)
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@Column
	@JsonView(Summary.class)
	private String name;
	
	@Column
	@JsonView(Summary.class)
	private String code;
	
	@Column
	private String dateFormat;
	
	@Column
	private String timeFormat;
	
	public Long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public String getCode() {
		return code;
	}

	public String getDateFormat() {
		return dateFormat;
	}

	public String getTimeFormat() {
		return timeFormat;
	}
}

File: src/main/kotlin/com/pancisin/bookster/model/enums/Locale.kt
package com.pancisin.bookster.model.enums

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonView
import com.pancisin.bookster.models.views.Summary

@JsonFormat(shape = JsonFormat.Shape.OBJECT)
enum class Locale private constructor(
  @JsonProperty("name")
  val prop: String,
  val code: String,
  val dateFormat: String,
  val timeFormat: String
) {
  sk("sk", "locale.sk", "DD.MM.YYYY", "HH:mm"), en("en", "locale.en", "MM/DD/YYYY", "H mm a")
}
